gen/generate_test.go

```diff
@@ -79,6 +79,62 @@ func TestGenerateWithRelativePaths(t *testing.T) {
 	}
 }
 
+func TestGenerateWithHyphenPaths(t *testing.T) {
+	outputDir, err := ioutil.TempDir("", "thriftrw-generate-test")
+	require.NoError(t, err)
+	defer os.RemoveAll(outputDir)
+
+	thriftRoot, err := os.Getwd()
+	require.NoError(t, err)
+
+	tests := []struct {
+		name          string
+		filepath      string
+		generateError string
+		compileError  string
+	}{
+		{
+			name:         "include with hyphens error",
+			filepath:     "internal/tests/thrift/nestedfiles_error/include_hyphen_files.thrift",
+			compileError: "cannot include hyphenated Thrift files",
+		},
+		{
+			name:     "normalization - hyphen files as top level file allowed",
```

Comment:
can use `/` to separate the test name to match subtests format.

```suggestion
			name:     "normalization/hyphen files as top level file allowed",
```